src/server/pkg/localcache/cache.go

```diff
@@ -0,0 +1,97 @@
+package localcache
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"sync"
+)
+
+// Cache is a simple unbounded disk cache and is safe for concurrency.
+type Cache struct {
+	root string
+	keys map[string]struct{}
```

Comment:
We tend to prefer using `map[string]bool` for sets in our code base which allows you to do:
```
if c.keys[key] {
  ...
}
```
rather than having to extract an `ok` parameter.

Reply:
Sure, I have a habit of defaulting to the map[string]struct{} data structure because of the small memory savings, but the readability might be more useful.